### ISSUE TITLE
feature request: Robbins-Monro type learning rate decay

### ISSUE THREAD

**Issue by jvishnuvardhan** [keras-team/keras#15608]
Note: Moving this from https://github.com/tensorflow/tensorflow/issues/20644
Keras-team/keras will evaluate this feature. Thanks

I was wondering if there is any appetite for a [Robbins-Monro](https://en.wikipedia.org/wiki/Stochastic_approximation#Robbins%E2%80%93Monro_algorithm) type learning rate decay in tensorflow? The decay would be roughly (a more general solution is implemented at the bottom):

```python
decayed_learning_rate = learning_rate * (global_step + 1) ^ (-decay_rate) 
```

As far as I can tell it is not already implemented in tensorflow, which surprised me since I think this is the learning rate decay rate required for theoretical convergence using the Adam optimizer in Section 4 of [the paper](https://arxiv.org/pdf/1412.6980.pdf), which has:

```python
alpha_t = alpha / sqrt(t)
```

which is the same as the first equation with `decay_rate = 0.5`, and I assume they start at `t = 1` while tensorflow starts with `global_step = 0`.

I have an implementation I have been using (mostly copied from the [already implemented ones](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/training/learning_rate_decay.py)):

```python
def robbins_monro_decay(learning_rate,
                        global_step,
                        decay_steps,
                        decay_rate,
                        staircase=False,
                        name=None):
    """A Robbins-Monro type decay"""

    if global_step is None:
        raise ValueError("global_step is required for robbins_monro_decay.")

    with tf.name_scope(
            name,
            "RobbinsMonroDecay",
            [learning_rate, global_step, decay_steps, decay_rate]) as name:

        learning_rate = tf.convert_to_tensor(
            learning_rate, name="learning_rate")
        dtype = learning_rate.dtype
        decay_steps = tf.cast(decay_steps, dtype)
        decay_rate = tf.cast(decay_rate, dtype)
        global_step = tf.cast(global_step, dtype)
        p = global_step / decay_steps

        if staircase:
            p = tf.floor(p)

        return tf.multiply(
            learning_rate, tf.pow(p + 1, -decay_rate), name=name)
```

I can make a full pull request if that would be useful? I would be more than happy to add some additional documentation/code or change any of the code/naming/whatever.

Thanks!

**Comment by old-school-kid** [comment]
Should this be moved to tensorflow-addons? @bhack

**Comment by SuryanarayanaY** [comment]
Hello, Thank you for reporting an issue.

We're currently in the process of migrating the new Keras 3 code base from keras-team/keras-core to keras-team/keras.
Consequently, This issue may not be relevant to the Keras 3 code base . After the migration is successfully completed, feel free to reopen this Issue at keras-team/keras if you believe it remains relevant to the Keras 3 code base. If instead this Issue is a bug or security issue in legacy tf.keras, you can instead report a new issue at [keras-team/tf-keras](https://github.com/keras-team/tf-keras/issues), which hosts the TensorFlow-only, legacy version of Keras.

To know more about Keras 3, please read https://keras.io/keras_core/announcement/
